Util/Xamarin.AndroidBinderator/Xamarin.AndroidBinderator/Engine.cs

```diff
@@ -134,142 +134,62 @@ static async Task DownloadArtifacts(BindingConfig config, Dictionary<string, Pro
-					sb.AppendLine($"OpenLibraryFile failed for: {mavenArtifact.GroupId}, {mavenArtifact.ArtifactId}, {version}");
-					sb.AppendLine($"Message");
-					sb.AppendLine($"{exc.Message}");
-					Trace.WriteLine(sb.ToString());
 				}
 
-				if (config.DownloadJavaSourceJars)
-				{
-					try
-					{
-						using (var astrm = await maven.OpenArtifactSourcesFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version))
-						using (var sw = File.Create(sourcesFile))
-							await astrm.CopyToAsync(sw);
-					}
-					catch (System.Exception exc)
-					{
-						StringBuilder sb = new StringBuilder();
-						sb.AppendLine($"DownloadJavaSourceJars failed for: {mavenArtifact.GroupId}, {mavenArtifact.ArtifactId}, {version}");
-						sb.AppendLine($"Message");
-						sb.AppendLine($"{exc.Message}");
-						Trace.WriteLine(sb.ToString());
-					}
-				}
+				var sourcesFile = Path.Combine(artifactDir, config.DownloadExternalsWithFullName ? $"{mavenArtifact.GroupId}.{mavenArtifact.ArtifactId}-sources.jar"
+					: $"{mavenArtifact.ArtifactId}-sources.jar");
 
-				if (config.DownloadPoms)
-				{
-					try
-					{
-						using (var astrm = await maven.OpenArtifactPomFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version))
-						using (var sw = File.Create(sourcesFile))
-							await astrm.CopyToAsync(sw);
-					}
-					catch (System.Exception exc)
-					{
-						StringBuilder sb = new StringBuilder();
-						sb.AppendLine($"DownloadPoms failed for: {mavenArtifact.GroupId}, {mavenArtifact.ArtifactId}, {version}");
-						sb.AppendLine($"Message");
-						sb.AppendLine($"{exc.Message}");
-						Trace.WriteLine(sb.ToString());
-					}
+				if (config.DownloadJavaSourceJars) {
+					var source_func = new Func<Task<Stream>>(() => maven.OpenArtifactSourcesFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version));
+					await TryDownloadFile(source_func, sourcesFile, ErrorLevel.Info);
 				}
 
-				if (config.DownloadJavaDocJars)
-				{
-					try
-					{
-						// using (var astrm = await maven.OpenArtifactDocsFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version))
-						// using (var sw = File.Create(sourcesFile))
-						// 	await astrm.CopyToAsync(sw);
-					}
-					catch(System.Exception exc)
-					{
-						StringBuilder sb = new StringBuilder();
-						sb.AppendLine($"DownloadJavaDocJars failed for: {mavenArtifact.GroupId}, {mavenArtifact.ArtifactId}, {version}");
-						sb.AppendLine($"Message");
-						sb.AppendLine($"{exc.Message}");
-						Trace.WriteLine(sb.ToString());
-					}
+				if (config.DownloadPoms) {
+					var pom_func = new Func<Task<Stream>>(() => maven.OpenArtifactPomFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version));
+					await TryDownloadFile(pom_func, sourcesFile, ErrorLevel.Info);
 				}
 
+				if (config.DownloadJavaDocJars) {
+					//var javadoc_func = new Func<Task<Stream>>(() => maven.OpenArtifactDocsFile(mavenArtifact.GroupId, mavenArtifact.ArtifactId, version));
```

Comment:
I think this cane be uncommented. I added javadoc support to MavenNet 

https://github.com/Redth/MavenNet/pull/6/files

Reply:
I updated MavenNet and now this function is available, however it is also writing to `sourcesFile` so the file will get overwritten in the next step.

What are the desired filenames for `OpenArtifactSourcesFile`/`OpenArtifactPomFile`/`OpenArtifactDocsFile`/`OpenMavenMetadataFile` and I will update this PR to write them correctly to disk.

Reply:
Well...

Google's naming (google repo)

```
$"{artifact_id}-{artifact_version}.aar"
$"{artifact_id}-{artifact_version}.jar"
$"{artifact_id}-{artifact_version}.pom"
$"{artifact_id}-{artifact_version}-sources.jar"
$"{artifact_id}-{artifact_version}-javadoc.jar"
$"{artifact_id}-{artifact_version}.module"
```

though module (metadata file) is actually JSON file, so you could add `.json` extension

Maven Central (Sonatype) differs for metadata

https://repo1.maven.org/maven2/com/google/crypto/tink/tink-android/

https://repo1.maven.org/maven2/com/google/crypto/tink/tink-android/maven-metadata.xml

but main files are the same:

```
$"{artifact_id}-{artifact_version}.aar"
$"{artifact_id}-{artifact_version}.jar"
$"{artifact_id}-{artifact_version}.pom"
$"{artifact_id}-{artifact_version}-sources.jar"
$"{artifact_id}-{artifact_version}-javadoc.jar"
```

HTH